### ISSUE TITLE
Remove all memory leaks in unit tests

### ISSUE THREAD

**Issue by marcusnaslund** [magic-lang/ooc-kean#1699]
This helps a lot when it comes to bug fixing.

- [x] `base`
- [x] `cli`
- [x] `collections` (only `HashDictionary` remains - see #694 )
- [x] `concurrent`
- [x] `draw`
- [x] `geometry`
- [x] `io`
- [x] `math`
- [x] `net`
- [x] `plot`
- [x] `system`
- [x] `unit`
- [x] `uri`

**Comment by marcusnaslund** [comment]
Updated with status after #1703

**Comment by marcusnaslund** [comment]
Leaks in concurrent fixed as of #1714 and so the only remaining leak is in `HashDictionary` which has a separate issue. Yay!
